### PR TITLE
refactor(source): refactor some code of source

### PR DESCRIPTION
**Pull request by waruto210**
I hereby agree to the terms of the [Singularity Data, Inc. Contributor License Agreement](https://gist.github.com/skyzh/0663682a70b0edde7ae991492f2314cb#file-s9y_cla).

## What's changed and what's your intention?

**This section will be used as the commit message. Please do not leave this empty!**

- refactor `FsSourceExecutor`, make it more unified with `SourceExecutor`.
- add `SourceMetric` statistics for s3 reader.
- rename some V2 structs
- refine some code

## Checklist

- [x] I have written necessary rustdoc comments
- [x] I have added necessary unit tests and integration tests
~- [] I have added fuzzing tests or opened an issue to track them. (Optional, recommended for new SQL features).~
- [x] All checks passed in `./risedev check` (or alias, `./risedev c`)

## Refer to a related PR or issue link (optional)
#7032

### CODE REVIEW

File: src/connector/src/source/mod.rs [hunk 1]
```diff
@@ -20,12 +20,29 @@ pub mod filesystem;
...
 pub use google_pubsub::GOOGLE_PUBSUB_CONNECTOR;
 pub use kafka::KAFKA_CONNECTOR;
 pub use kinesis::KINESIS_CONNECTOR;
+use risingwave_common::catalog::TableId;
 
 pub use crate::source::nexmark::NEXMARK_CONNECTOR;
 pub use crate::source::pulsar::PULSAR_CONNECTOR;
+
+#[derive(Clone, Copy, Debug, Default)]
+pub struct SourceContext {
```
Comment (TennyZhuang) [inline src/connector/src/source/mod.rs:36]: Is `SourceInfo` a better name?

If you think it's a type of `Context`, you can also store `SourceMetrics` here and store the whole Context in an `Arc`.

**Review by TennyZhuang — APPROVED** [review]
Rest LGTM

File: src/connector/src/source/mod.rs (hunk 1 above)
Comment (waruto210) [reply src/connector/src/source/mod.rs]: Renamed it to `SourceInfo`

File: src/connector/src/source/base.rs [hunk 2]
```diff
@@ -107,6 +107,8 @@ pub trait SplitReaderV2: Sized {
         properties: Self::Properties,
         state: Vec<SplitImpl>,
         parser_config: ParserConfig,
+        metrics: Arc<SourceMetrics>,
+        source_context: SourceInfo,
```
Comment (xx01cyx) [inline src/connector/src/source/base.rs:111]: The variable name may need renaming as well 🤔 (ditto for all other variables)

**Review by tabVersion — APPROVED** [review]
LGTM

File: src/connector/src/source/base.rs (hunk 2 above)
Comment (waruto210) [reply src/connector/src/source/base.rs]: Done, thanks for reminding!

**Comment by waruto210** [comment]
> The implementations of `SourceDesc` and `FsSourceDesc` are very much alike. Is there a more elegant way to do this? cc @BugenZhao

This is intentional, in case the logic of the two Sources becomes more different in the future.

File: src/stream/src/from_proto/source.rs [hunk 3]
```diff
@@ -83,40 +85,33 @@ impl ExecutorBuilder for SourceExecutorBuilder {
...
-            let is_s3 = source
+            let connector = source
                 .properties
                 .get("connector")
-                .map(|s| s.to_lowercase())
-                .unwrap_or_default()
-                .eq("s3");
-            if is_s3 {
+                .map(|c| c.to_ascii_lowercase())
+                .unwrap();
```
[the remark this answers is not in the document]
Comment (waruto210) [reply src/stream/src/from_proto/source.rs:100]: I'll use `unwrap_or_default` instead.

**Pull request merged**